orttraining/orttraining/python/orttraining_pybind_state.cc

```diff
@@ -812,42 +824,58 @@ void addObjectMethodsForTraining(py::module& m, ExecutionProviderRegistrationFn
-                                          const std::unordered_set<std::string>& y_node_arg_names,
-                                          const std::unordered_set<std::string>& x_node_arg_names,
-                                          const std::string loss_node_arg_name) {
-                          std::shared_ptr<Model> model;
-                          auto logger_ptr = std::make_unique<logging::Logger>(logging::LoggingManager::DefaultLogger());
-                          logger_ptr->SetSeverity(logging::Severity::kINFO);
-                          ONNX_NAMESPACE::ModelProto model_proto;
-                          std::istringstream model_istream(serialized_model);
-                          ORT_THROW_IF_ERROR(Model::Load(model_istream, &model_proto));
-                          ORT_THROW_IF_ERROR(Model::Load(model_proto, model, nullptr, *logger_ptr));
-                          GradientGraphConfiguration gradient_graph_config{};
-                          gradient_graph_config.set_gradients_as_graph_outputs = true;
-                          // Save some objects, otherwise they get lost.
-                          auto gradient_graph_config_ptr = std::make_unique<GradientGraphConfiguration>(gradient_graph_config);
-
-                          auto builder = std::make_unique<GradientGraphBuilder>(
-                              &model->MainGraph(),
-                              y_node_arg_names,
-                              x_node_arg_names,
-                              loss_node_arg_name,
-                              *gradient_graph_config_ptr,
-                              *logger_ptr);
-
-                          return std::make_unique<PyGradientGraphBuilder>(std::move(builder), std::move(model), std::move(logger_ptr), std::move(gradient_graph_config_ptr));
-                        }))
+  gradient_graph_builder
+      .def(py::init([](const py::bytes& serialized_model,
+                       const std::unordered_set<std::string>& y_node_arg_names,
+                       const std::unordered_set<std::string>& x_node_arg_names,
+                       const std::string loss_node_arg_name,
+                       PySessionOptions* options) {
+             std::shared_ptr<CustomRegistry> custom_registry;
+             IOnnxRuntimeOpSchemaRegistryList local_registries;
+             if (options && !options->custom_op_domains_.empty()) {
+               // Register all custom op domains that will be needed for the session
+               ORT_THROW_IF_ERROR(onnxruntime::CreateCustomRegistry(options->custom_op_domains_, custom_registry));
+               local_registries.push_back(custom_registry->GetOpschemaRegistry());
+             }
+
+             std::shared_ptr<Model> model;
+             auto logger_ptr = std::make_unique<logging::Logger>(logging::LoggingManager::DefaultLogger());
+             logger_ptr->SetSeverity(logging::Severity::kINFO);
```

Comment:
if we pass session options, shall we parse log_severity_level from it and use it instead of hard coding to info?